### PR TITLE
add unit tests

### PR DESCRIPTION
**Pull request by Bee-lee**
Work in progress. Need to discuss.
Re-created PR so its draft because WIP keyword doesnt work on GitHub

### CODE REVIEW

File: responses/responses_test.go [hunk 1]
```diff
@@ -15,3 +15,117 @@ limitations under the License.
...
+		"data":   mock_payload,
+		"status": "ok",
+	}
+
+	t.Log(response)
+	t.Log(expected_response)
+	// why aren't they equal???
+	if equal := reflect.DeepEqual(expected_response, response); !equal {
+		t.Errorf("Expected response %v.", expected_response)
+	}
```
Comment (Bee-lee) [inline responses/responses_test.go]: Guys @tisnik @bond95 I don't understand this. The printed response and expected response look the same, but the DeepEqual check says they arent equal. Am I missing something?

Comment (tisnik) [reply responses/responses_test.go]: I'd need to debug it, lemme check

Comment (Bee-lee) [reply responses/responses_test.go]: I solved it a little differently as not to waste too much time on just unit tests.
We can discuss tomorrow

File: responses/responses_test.go [hunk 2]
```diff
@@ -15,3 +15,159 @@ limitations under the License.
...
+
+		if equal := reflect.DeepEqual(response, expected); !equal {
+			t.Errorf("Expected response %v.", expected)
+		}
+	}
+}
+
+// Test BuildResponse that returns simple map with key "status" and given value
+func TestBuildResponse(t *testing.T) {
+	status_str := "I'm a teapot"
```
Comment (tisnik) [inline responses/responses_test.go]: me too :)

**Comment by Bee-lee** [comment]
@tisnik is it OK like this? :)

**Comment by Bee-lee** [comment]
Sure, of course, I would have tagged dmitri myself (next time) but I don't know the github usernames yet

**Pull request merged**